### ISSUE TITLE
Unit testing

### ISSUE THREAD

**Issue by KevinGrandon** [mvines/ci-gate#38]
Would you be open to adding a unit testing framework for this repo? Given the nature of this logic, my recommendation would be going with Jest for it's powerful mocking abilities. A unit testing framework would allow for more contributions, as we could make them with confidence.

**Comment by mvines** [comment]
Absolutely!  It's kinda terrifying that there are no tests right now, I'm using this in production at https://github.com/solana-labs/.  It's a little embarrassing honestly.

**Comment by KevinGrandon** [comment]
Awesome, glad you're still using this and Buildkite. I'll try to make some contributions since we're going to be using it for most of our open source stuff at Uber for the foreseeable future :)
